coding-concepts.js

```diff
@@ -6,49 +6,52 @@
 // Then, uncomment the console.log() and verify the output. Briefly explain why your initial answer was correct or incorrect.
 
 // --------------------INSTRUCTOR EXAMPLE: What will this log?
-const colors = ["tangerine", "magenta", "lilac", "daffodil"]
+// const colors = ["tangerine", "magenta", "lilac", "daffodil"]
 // console.log(colors.push("indigo"))
 
 // a) Your answer:
 // b) Verify and explain:
 
 // --------------------1) What will this log?
 
-const cohort = "LEARN 2023"
+// const cohort = "LEARN 2023"
 // console.log(cohort.length)
 
-// a) Your answer:
-// b) Verify and explain:
-
-// --------------------2) What will this log?
+// a) Your answer: 9
+// b) Verify and explain: Incorrect, I thought it was LEARN 2023
+// But I was incorrect.                               0123456789
+// I forgot .length was different to index value, so there would be no 0.
+// // --------------------2) What will this log?
 
-const greeting = "Hello World!"
+// const greeting = "Hello World!"
 // console.log(greeting[4])
 
-// a) Your answer:
-// b) Verify and explain:
+// // a) Your answer: o
+// // b) Verify and explain: Correct! The index of the string. 0 is H. Counting upwards to 4 arives at o.
 
-// --------------------3) What will this log?
+// // --------------------3) What will this log?
 
-const languages = ["JavaScript", "Ruby", "Python", "C++"]
-const index = 1
+// const languages = ["JavaScript", "Ruby", "Python", "C++"]
+// const index = 1
 // console.log(languages[index])
 
-// a) Your answer:
-// b) Verify and explain:
+// // a) Your answer:Ruby 
+// // b) Verify and explain: Correct! the index of the array. 0 = javascript, counting to 1 = Ruby.
 
-// --------------------4) What will this log?
+// // --------------------4) What will this log?
 
-const weekendDays = ["saturday", "sunday"]
+// const weekendDays = ["saturday", "sunday"]
 // console.log(weekendDays.toUpperCase())
 
-// a) Your answer:
-// b) Verify and explain:
+// // a) Your answer: SATURDAYSUNDAY
+// // b) Verify and explain: Incorrect, toUpperCase is not a function error. It should be outside of the console log and inside brackets () in order to work. Example:
+//  if/else(weekendDays.toUpperCase)
+//  console.log(weekendDays)
 
-// --------------------5) What will this log?
+// // --------------------5) What will this log?
 
 const dataTypes = ["number", "string", "Boolean", "undefined"]
-// console.log(typeof dataTypes.length)
+console.log(typeof dataTypes.length)
 
-// a) Your answer:
-// b) Verify and explain:
+// // a) Your answer: Some type of error due to typeof?
+// // b) I thought camelCase was required and was unsure typeof was a valid built in method for Arrays/.js code. But 4 is returned due to the 4 different types of primitivie data types listed in the strings, or just listing the amount of strings? 
```

Comment:
The `typeof` operator is used to return the data type of the data that follows it. In this case, the data that follows `typeof` is `dataTypes.length`. If we just do a console.log on `dataTypes.length`, we will get 4, because there are 4 things in the array. Then, `typeof` is looking at that 4 and seeing that it is a number data type, which is why when you run this code, it will return `number`.